src/target/riscv/riscv.h

```diff
@@ -2,66 +2,176 @@
 #define RISCV_H
 
 #include "opcodes.h"
+#include "gdb_regs.h"
+
+/* The register cache is staticly allocated. */
+#define RISCV_MAX_HARTS 32
+#define RISCV_MAX_REGISTERS 5000
 
 extern struct target_type riscv011_target;
 extern struct target_type riscv013_target;
 
 /*
  * Definitions shared by code supporting all RISC-V versions.
  */
+typedef uint64_t riscv_reg_t;
+
+enum riscv_hart_state {
+	RISCV_HART_UNKNOWN,
+	RISCV_HART_HALTED,
+	RISCV_HART_RUNNING,
+};
+
+enum riscv_halt_reason {
+	RISCV_HALT_INTERRUPT,
```

Comment:
we should probably call this 'haltreq' instead of interrupt, since interrupt is a sifive-ism